app/services/streaming/streaming.ts

```diff
@@ -982,6 +1061,308 @@ export class StreamingService
     }
   }
 
+  /**
+   * Part of a workaround to set recording quality on the frontend to avoid a backend bug
+   * @remark TODO: remove when migrate output and stream settings to new API
+   * Because of a bug with the new API and having the `Same as stream` setting, temporarily store and set
+   * the recording quality on the frontend
+   * @param mode - single output or dual output
+   */
+  beforeStartRecording(mode: 'single' | 'dual') {
```

Comment:
Here's another instance of the `'single' | 'dual'` type.

Reply:
Done